.github/workflow/multi-arch-build.yaml

```diff
@@ -0,0 +1,208 @@
+            # Assume some contents changed, always push latest testing.
+            FQIN="$SKOPEO_QUAY_REGISTRY/testing:latest"
+          elif [[ "${{ matrix.source }}" == 'upstream' ]]; then
+            # Assume some contents changed, always push latest upstream.
+            FQIN="$SKOPEO_QUAY_REGISTRY/upstream:latest"
+          else
+            echo "::error::Unknown matrix item '${{ matrix.source }}'"
+            exit 1
+          fi
+          echo "::warning::Pushing $FQIN"
+          echo "::set-output name=fqin::${FQIN}"
+          echo '::set-output name=push::true'
+
+      # This is substantially the same as the above step, except the
+      # $CONTAINERS_QUAY_REGISTRY is used and the "testing"
+      # flavor is never pushed.
+      - name: Generate containers reg. image FQIN(s)
+        if: matrix.source != 'testing'
+        id: containers_reg
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            VERSION='v${{ steps.sniff_test.outputs.version }}'
+            # workaround vim syntax-highlight bug: '
+            ALLTAGS=$(skopeo list-tags \
+                      docker://$CONTAINERS_QUAY_REGISTRY/skopeo | \
+                      jq -r '.Tags[]')
+
+            # New image? Push quay.io/containers/skopeo:vX.X.X and :latest
+            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
+              FQIN="$CONTAINERS_QUAY_REGISTRY/skopeo:$VERSION,$CONTAINERS_QUAY_REGISTRY/skopeo:latest"
+            else # Not a new version-tagged image, but contents may be updated
+              FQIN="$CONTAINERS_QUAY_REGISTRY/skopeo:latest"
+            fi
+          elif [[ "${{ matrix.source }}" == 'upstream' ]]; then
+            FQIN="$CONTAINERS_QUAY_REGISTRY/skopeo:latest"
+          else
+            echo "::error::Unknown matrix item '${{ matrix.source }}'"
+            exit 1
+          fi
+          echo "::warning::Pushing $FQIN"
+          echo "::set-output name=fqin::${FQIN}"
+          echo '::set-output name=push::true'
+
+      - name: Define LABELS multi-line env. var. value
+        run: |
+          # This is a really hacky/strange workflow idiom, required
+          # for setting multi-line $LABELS value for consumption in
+          # a future step.
+          # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings
+          cat << EOF | tee -a $GITHUB_ENV
+          LABELS<<DELIMITER
+          org.opencontainers.image.source=https://github.com/${{ github.repository }}.git
+          org.opencontainers.image.revision=${{ github.sha }}
+          org.opencontainers.image.created=$(date -u --iso-8601=seconds)
+          DELIMITER
+          EOF
+
+      # Separate steps to login and push for skopeo and containers quay
+      # repositories are required, because 2 sets of credentials are used and `docker
+      # login` as well as `skopeo login` do not support having 2 different
+      # credential sets for 1 registry.
```

Comment:
Really?  When I login to my account on quay.io using skopeo_login, I can push to either the skopeo or containres repos.  For that matter, a couple of others too.

Reply:
In that case you're using the same credentials for both pushes but it is correct that auth files allow for storing only 1 credential set for a given registry.

Reply:
Yep, and this is enforced even more for "robot" accounts, as they're forced to a single repository namespace.